lib/Doctrine/ODM/MongoDB/Iterator/CachingIterator.php

```diff
@@ -121,43 +117,35 @@ public function valid() : bool
      */
     private function exhaustIterator() : void
     {
-        while (! $this->iteratorExhausted) {
+        while ($this->iterator) {
             $this->next();
         }
 
         $this->iterator = null;
     }
 
-    private function getIterator() : Generator
-    {
-        if ($this->iterator === null) {
-            throw new RuntimeException('Iterator has already been destroyed');
-        }
-
-        return $this->iterator;
-    }
-
     /**
      * Stores the current item in the cache.
      */
     private function storeCurrentItem() : void
     {
-        $key = $this->getIterator()->key();
+        $key = $this->iterator->key();
 
         if ($key === null) {
             return;
         }
 
-        $this->items[$key] = $this->getIterator()->current();
+        $this->items[$key] = $this->iterator->current();
     }
 
     private function wrapTraversable(Traversable $traversable) : Generator
     {
         foreach ($traversable as $key => $value) {
+            $this->items[$key] = $value;
```

Comment:
I'd rather reuse `storeCurrentItem` here:
```suggestion
            $this->storeCurrentItem();
```

That said, I'm not sure why this refactoring is necessary, could you elaborate on that please?

Reply:
Reverted changes to `CachingIterator` in 3dd3423525672225acbebd4c4ebc72d5ff326edd